pyomo/gdp/plugins/hull.py

```diff
@@ -714,7 +726,8 @@ def _transform_block_components( self, block, disjunct, var_substitute_map,
         # transformation blocks for nested disjunctions, so that we don't have
         # duplicate references.
         varRefBlock = disjunctBlock.localVarReferences
-        for v in block.component_objects(Var, descend_into=Block, active=None):
+        for v in block.component_data_objects(Var, descend_into=Block,
+                                              active=None):
             varRefBlock.add_component(unique_component_name(
                 varRefBlock, v.getname(fully_qualified=True,
                                        name_buffer=NAME_BUFFER)), Reference(v))
```

Comment:
@jsiirola, this actually introduces a bug because `Reference(v)` fails when v is a `_GeneralVarData`, so it dies if you have an indexed variable declared on the Disjunct. But the reason I made the change was because without it, with the other changes, some transformed models don't pprint: They die in the `localVarReferences' block... I'm confused.

Reply:
OK, it's hitting #2039. I think we can just check that we are actually making a Reference to *something* and skip it if not, since we know we aren't going to add to the containers after we make the Reference.